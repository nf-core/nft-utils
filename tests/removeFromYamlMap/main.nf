workflow {
    channel
        .of(
            """
            Workflow:
                Pipeline: 1.0.0
                Nextflow: ${workflow.nextflow.version}
            Workflow2:
                TEMPLATE: 1.0.0
                Pipeline: 1.0.0
            """.stripIndent().trim()
        )
        .collectFile(storeDir: "${params.outdir}/pipeline_info", name: 'nf_core_pipeline_software_mqc_versions.yml', sort: true, newLine: true)
}
