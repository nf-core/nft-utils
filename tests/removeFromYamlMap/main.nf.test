nextflow_pipeline {

    name "Test removeFromYamlMap"
    script "./main.nf"
    tag "removeFromYamlMap"

    test("removeFromYamlMap") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assert snapshot(removeFromYamlMap("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml", "Workflow", "Nextflow")).match()
        }
    }

    test("removeFromYamlMap only key1") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assert snapshot(removeFromYamlMap("$outputDir/pipeline_info/*_versions.yml", "Workflow")).match()
        }
    }
}
