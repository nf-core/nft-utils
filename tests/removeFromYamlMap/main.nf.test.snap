{
    "removeFromYamlMap only key1": {
        "content": [
            {
                "Workflow2": {
                    "TEMPLATE": "1.0.0",
                    "Pipeline": "1.0.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-05T13:51:51.69206463"
    },
    "removeFromYamlMap": {
        "content": [
            {
                "Workflow": {
                    "Pipeline": "1.0.0"
                },
                "Workflow2": {
                    "TEMPLATE": "1.0.0",
                    "Pipeline": "1.0.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-05T13:26:18.982486633"
    }
}